tests/test_cli.py

```diff
@@ -40,7 +40,6 @@ def test_add_host(self, caplog, host):
         cli(["-c", config, "add_host", "-n", subsystem, "-t", host])
         assert "Failed to add" not in caplog.text
 
-    @pytest.mark.parametrize("listener", listener_list)
     def test_create_listener(self, caplog, listener):
```

Comment:
The last thing (I hope), please remove`listener` arg here and in `test_delete_listener`. BTW, have you ran the tests? Do they pass?

Reply:
I only manually tested the gateway commands. I haven't run the testing module before, and I haven't found any relevant instructions in README. Additionally, I'm not very familiar with Python. Could you please teach me how to run the testing module.

Reply:
Assuming that you have the gateway properly configured and running and in  a clean state (no bdev, namespaces, etc are configured) [1], and you are using "rbd" pool, to run the test you need:

1) create an image named "iscsidevimage" (you may find the image name in test_cli.py, and change if you don't like it, and also the pool name if it is not "rbd"):

  `rbd create iscsidevimage -s 1G`

2) run the test with the help of pytest:

 `pytest tests/test_cli.py`
  
 You may run all tests, not only cli with the command:
 
   `pytest tests`
   
 but for the `test_multi_gateway.py` to succeed you might need some additional tweaking for your environment (I remember I needed though I don't remember what exactly right now).
  
[1] If the state of your gateway is not clean before running the test, to clean it, you may stop the gateway, delete
nvmeof.state object with the command `rados -p rbd rm nvmeof.state`, and start the gateway again.

Reply:
Thanks a lot. After some attempts, I finally succeeded.
```
root@cephs:~/git/ceph-nvmeof# pytest-3 tests/test_cli.py
======================================================================================================= test session starts ========================================================================================================
platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0
rootdir: /root/git/ceph-nvmeof
collected 13 items

tests/test_cli.py .............                                                                                                                                                                                              [100%]

==================================================================================================== 13 passed in 0.30 seconds =====================================================================================================
```